pkg/virt-handler/vm.go

```diff
@@ -2881,3 +2892,26 @@ func nodeHasHostModelLabel(node *k8sv1.Node) bool {
 	}
 	return false
 }
+
+func (d *VirtualMachineController) reportDedicatedCPUSetForMigratingVMI(vmi *v1.VirtualMachineInstance) error {
+	baseCPUSetPath := "/proc/1/root/sys/fs/cgroup/cpuset"
+	isoRes, err := d.podIsolationDetector.Detect(vmi)
+	if err != nil {
+		return err
+	}
+
+	cpusetFilePath := filepath.Join(baseCPUSetPath, isoRes.Slice(), "cpuset.cpus")
```

Comment:
Cgroup cpuset path differs between `cgroup v1` and `cgroup v2`. Please take a look at`pkg/virt-handler/cgroup/cgroup.go` package. I think a function similar to `CPUSetPath()` but with `slice` as parameter is what you need here: https://github.com/kubevirt/kubevirt/blob/3c96a02eac2ee50b9571be47aeaa36dc0e74cc3a/pkg/virt-handler/cgroup/cgroup.go#L50-L52

Also FYI there is a cgroup `v2` presubmit CI job that can be triggered manually: `pull-kubevirt-e2e-k8s-1.20-cgroupsv2`. I think it would make sense to run it here.

Reply:
Thanks! I'll take a look at that

Reply:
With https://github.com/kubevirt/kubevirt/pull/6251 this issue should now be much easier to fix. We already get all the cpu information with `VirtualMachineOptions` which we can pass through for the sync and prepare calls from virt-handler. virt-launcher can then simply feed that to the new pool policy and apply the resulting mapping 1:1 to this XML fragment.